crates/autopilot/src/run_loop.rs

```diff
@@ -493,14 +495,66 @@ impl RunLoop {
     }
 }
 
+/// Checks whether or not an order's limit price is outside the market price
+/// specified by the quote.
+///
+/// Note that this check only looks at the order's limit price and the market
+/// price and is independent of amounts or trade direction.
+pub fn is_order_outside_market_price(
```

Comment:
Instead of duplicating this function you can take the original one and make it reusable for this use case as well.

Reply:
It is unfortunately not possible. 

The original function takes shared::order_quoting::Quote as argument.
shared::order_quoting::Quote is filled by the QuoteStoring which is read by crates/database/src/quotes.rs::get method. Not that the function is using quotes table.

On the other hand,  @sunce86 recommended me to use crates/database/src/orders.rs::read_quote (which returns database::orders::Quote).

Reply:
You should be able to refactor `is_order_outside_market_price` to use only `U256` values and than reuse it in autopilot. I forsee the autopilot function will eventually be dropped and the check be done only in orderbook and populating the order class as @fleupold suggested (but can't be done now).

Reply:
that's correct. will do that !

Reply:
refactored